### ISSUE TITLE
millis() function inside ISR not working

### ISSUE THREAD

**Issue by smdll** [GreyGnome/PinChangeInt#6]
An millis() function inside ISR function is not working like usual, instead it jamms the whole program until manually reset the board.

**Comment by GreyGnome** [comment]
Yes, millis uses an interrupt to perform its updates, so if you're in an
isr the interrupts are shut off and no updates, so you are stuck.

You should not put delays of any kind in an isr. If you want to, you are
probably thinking about your problem incorrectly. You need to get out of
the isr as soon as possible, otherwise you are crippling your processor.
The thing to do is to have the isr set a flag which you check in your main
loop. This is a little tricky but not very hard. You have to make sure your
flag is not optimized. You mark it as volatile. See some of the examples.

I hope that helps. Let me know if you have any further questions.

**Comment by smdll** [comment]
Okay... Thanks.
